### ISSUE TITLE
Request Params: Query takes precedance over path

### ISSUE THREAD

**Issue by pedromorgan** [revel/revel#1095]
Here's the route

```
GET /ajax/ags/abbrev/:code     Ags.Abbrev
```

and controller

``` go
func (c Ags) Abbrev(code string) revel.Result {
    fmt.Println("arg=", code)
    fmt.Println("query=", c.Request.URL.Query().Get("code"))
}
```

and the url + output

```
http://localhost:9000/ajax/ags/abbrev/CMPG_TYPE?code=CMPG_TYPEsss

GET /ajax/ags/abbrevs
arg= CMPG_TYPEsss
query= CMPG_TYPEsss

```

Whats scared me is the somehow the `?query`  is overriding the "path" extraction.

from here its says their "all in one params" http://revel.github.io/manual/parameters.html
which is fair, but is this not dangerous ??

The expected behaviour  would be that the "path" part would override the "query part", or am i missing something ??

**Comment by pedromorgan** [comment]
So digging down for curious.. Think its here calcValues/.. Should be called "parseValues"
https://github.com/revel/revel/blob/master/params.go#L75

and probalby here is clash.. as we expect the "path" to be set after ALL query stuff etc..
https://github.com/revel/revel/blob/master/params.go#L83

Anyway, they are sepeate things.. after looking at code..
But we HAVE to make this clear

**Comment by pedromorgan** [comment]
```csv
## Hot Zone
GET /reveldevil/params                        RevelDevil.ParaHell
GET /reveldevil/params/:code                  RevelDevil.ParaHell
GET /reveldevil/params/:code/:hell            RevelDevil.ParaHell
GET /reveldevil/params/:code/:hell/:haven     RevelDevil.ParaHell
```

and the controllers
```
func (c RevelDevil) ParaHell(code string) revel.Result {

	stuff := make(map[string]interface{})
	stuff["0_url="] =  c.Request.URL.String()
	stuff["1_aqquiring a param named="] = "code"
	stuff["2_revel_params="] = c.Params
	stuff["3_controller_method_arg="] = code
	stuff["4_revel_values="] = c.Params.Values.Get("code")
	stuff["5_raw_query_arg"] = c.Request.URL.Query().Get("code")
	stuff["6_revel_route/:code"] = c.Params.Route.Get("code")
	stuff["7_revel_query?code="] = c.Params.Query.Get("code")


	return c.RenderJson(stuff)
}
```
results =
```js
{
  "0_url=": "/reveldevil/params/YIPEE/flower?code=xxx\u0026code=xxx\u0026code=zzz",
  "1_aqquiring a param named=": "code",
  "2_revel_params=": {
    "Values": {
      "code": [
        "xxx",
        "xxx",
        "zzz",
        "YIPEE"
      ],
      "hell": [
        "flower"
      ]
    },
    "Fixed": null,
    "Route": {
      "code": [
        "YIPEE"
      ],
      "hell": [
        "flower"
      ]
    },
    "Query": {
      "code": [
        "xxx",
        "xxx",
        "zzz"
      ]
    },
    "Form": null,
    "Files": null
  },
  "3_controller_method_arg=": "xxx",
  "4_revel_values=": "xxx",
  "5_raw_query_arg": "xxx",
  "6_revel_route/:code": "YIPEE",
  "7_revel_query?code=": "xxx"
}
```

**Comment by pedromorgan** [comment]
Maybe code below should be final "loop", rewriting any previous values near
https://github.com/revel/revel/blob/master/params.go#L96

```go
	for k, v := range p.Route {
		values[k] = v
	}
```

**Comment by alkchr** [comment]
> from here its says their "all in one params" http://revel.github.io/manual/parameters.html
> which is fair, but is this not dangerous ??

@pedromorgan This is convenient but cannot be considered a good practice. And it [made](https://github.com/revel/revel/issues/503#issuecomment-65252825) #503 and later #809 possible.

**Comment by pedromorgan** [comment]
So #503 Explains this well

**Comment by shawncatz** [comment]
resolved by #1126
